server/services/user.service.js

```diff
@@ -84,6 +86,13 @@ export const addUser = async (user) => {
   try {
     const password = await hashPassword(user.password);
     const savedUser = await new User({ ...user, password, referralCode }).save();
+
+    if (user.referralCode) {
+      const referee = await getUserByReferralCode(user.referralCode).catch((error) => {
+        throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+      });
+      await addReferral({ userId: savedUser._id, refereeId: referee._id });
```

Comment:
This fails when I tested it. I believe it should be `referrerId`. Also can we have have a test that ensures that the referral table is created appropriate when they register.
```suggestion
      await addReferral({ userId: savedUser._id, referrerId: referrer._id });
```